// eslint-plugin-sonarjs@2.0.4/cjs/S6302/rule.js
"use strict";
/*
 * SonarQube JavaScript Plugin
 * Copyright (C) 2011-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// https://sonarsource.github.io/rspec/#/rspec/S6302/javascript
Object.defineProperty(exports, "__esModule", { value: true });
exports.rule = void 0;
const result_js_1 = require("../helpers/result.js");
const iam_js_1 = require("../helpers/aws/iam.js");
const index_js_1 = require("../helpers/index.js");
const meta_js_1 = require("./meta.js");
const MESSAGES = {
    message: 'Make sure granting all privileges is safe here.',
    secondary: 'Related effect',
};
exports.rule = (0, iam_js_1.AwsIamPolicyTemplate)(allPrivilegesStatementChecker, (0, index_js_1.generateMeta)(meta_js_1.meta, undefined, true));
function allPrivilegesStatementChecker(expr, ctx, options) {
    const properties = (0, result_js_1.getResultOfExpression)(ctx, expr);
    const effect = (0, iam_js_1.getSensitiveEffect)(properties, ctx, options);
    const action = getSensitiveAction(properties, options);
    if (effect.isMissing && action) {
        (0, index_js_1.report)(ctx, {
            message: MESSAGES.message,
            node: action,
        });
    }
    else if (effect.isFound && action) {
        (0, index_js_1.report)(ctx, {
            message: MESSAGES.message,
            node: action,
        }, [(0, index_js_1.toSecondaryLocation)(effect.node, MESSAGES.secondary)]);
    }
}
function getSensitiveAction(properties, options) {
    return getActionLiterals(properties, options).find(iam_js_1.isAnyLiteral);
}
function getActionLiterals(properties, options) {
    return properties.getProperty(options.actions.property).asStringLiterals();
}
