// eslint-plugin-sonarjs@2.0.4/cjs/decorated.js
"use strict";
/*
 * SonarQube JavaScript Plugin
 * Copyright (C) 2011-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.S6535 = exports.S131 = exports.S1534 = exports.S107 = exports.S1438 = exports.S3512 = exports.S6557 = exports.S6666 = exports.S6661 = exports.S6606 = exports.S4156 = exports.S6598 = exports.S4138 = exports.S6572 = exports.S3498 = exports.S3504 = exports.S6647 = exports.S6676 = exports.S1068 = exports.S905 = exports.S1763 = exports.S6747 = exports.S3696 = exports.S4327 = exports.S7060 = exports.S6679 = exports.S6571 = exports.S6440 = exports.S2814 = exports.S6544 = exports.S6660 = exports.S2189 = exports.S6788 = exports.S1116 = exports.S6643 = exports.S4023 = exports.S1186 = exports.S6551 = exports.S4275 = exports.S2430 = exports.S4084 = exports.S6853 = exports.S6749 = exports.S5254 = exports.S1788 = exports.S1105 = exports.S6844 = exports.S6827 = exports.S1077 = exports.S2376 = void 0;
exports.S2688 = void 0;
var index_js_1 = require("./S2376/index.js"); // accessor-pairs
Object.defineProperty(exports, "S2376", { enumerable: true, get: function () { return index_js_1.rule; } });
var index_js_2 = require("./S1077/index.js"); // alt-text
Object.defineProperty(exports, "S1077", { enumerable: true, get: function () { return index_js_2.rule; } });
var index_js_3 = require("./S6827/index.js"); // anchor-has-content
Object.defineProperty(exports, "S6827", { enumerable: true, get: function () { return index_js_3.rule; } });
var index_js_4 = require("./S6844/index.js"); // anchor-is-valid
Object.defineProperty(exports, "S6844", { enumerable: true, get: function () { return index_js_4.rule; } });
var index_js_5 = require("./S1105/index.js"); // brace-style
Object.defineProperty(exports, "S1105", { enumerable: true, get: function () { return index_js_5.rule; } });
var index_js_6 = require("./S1788/index.js"); // default-param-last
Object.defineProperty(exports, "S1788", { enumerable: true, get: function () { return index_js_6.rule; } });
var index_js_7 = require("./S5254/index.js"); // html-has-lang
Object.defineProperty(exports, "S5254", { enumerable: true, get: function () { return index_js_7.rule; } });
var index_js_8 = require("./S6749/index.js"); // jsx-no-useless-fragment
Object.defineProperty(exports, "S6749", { enumerable: true, get: function () { return index_js_8.rule; } });
var index_js_9 = require("./S6853/index.js"); // label-has-associated-control
Object.defineProperty(exports, "S6853", { enumerable: true, get: function () { return index_js_9.rule; } });
var index_js_10 = require("./S4084/index.js"); // media-has-caption
Object.defineProperty(exports, "S4084", { enumerable: true, get: function () { return index_js_10.rule; } });
var index_js_11 = require("./S2430/index.js"); // new-cap
Object.defineProperty(exports, "S2430", { enumerable: true, get: function () { return index_js_11.rule; } });
var index_js_12 = require("./S4275/index.js"); // no-accessor-field-mismatch
Object.defineProperty(exports, "S4275", { enumerable: true, get: function () { return index_js_12.rule; } });
var index_js_13 = require("./S6551/index.js"); // no-base-to-string
Object.defineProperty(exports, "S6551", { enumerable: true, get: function () { return index_js_13.rule; } });
var index_js_14 = require("./S1186/index.js"); // no-empty-function
Object.defineProperty(exports, "S1186", { enumerable: true, get: function () { return index_js_14.rule; } });
var index_js_15 = require("./S4023/index.js"); // no-empty-interface
Object.defineProperty(exports, "S4023", { enumerable: true, get: function () { return index_js_15.rule; } });
var index_js_16 = require("./S6643/index.js"); // no-extend-native
Object.defineProperty(exports, "S6643", { enumerable: true, get: function () { return index_js_16.rule; } });
var index_js_17 = require("./S1116/index.js"); // no-extra-semi
Object.defineProperty(exports, "S1116", { enumerable: true, get: function () { return index_js_17.rule; } });
var index_js_18 = require("./S6788/index.js"); // no-find-dom-node
Object.defineProperty(exports, "S6788", { enumerable: true, get: function () { return index_js_18.rule; } });
var index_js_19 = require("./S2189/index.js"); // no-infinite-loop
Object.defineProperty(exports, "S2189", { enumerable: true, get: function () { return index_js_19.rule; } });
var index_js_20 = require("./S6660/index.js"); // no-lonely-if
Object.defineProperty(exports, "S6660", { enumerable: true, get: function () { return index_js_20.rule; } });
var index_js_21 = require("./S6544/index.js"); // no-misused-promises
Object.defineProperty(exports, "S6544", { enumerable: true, get: function () { return index_js_21.rule; } });
var index_js_22 = require("./S2814/index.js"); // no-redeclare
Object.defineProperty(exports, "S2814", { enumerable: true, get: function () { return index_js_22.rule; } });
var index_js_23 = require("./S6440/index.js"); // rules-of-hooks
Object.defineProperty(exports, "S6440", { enumerable: true, get: function () { return index_js_23.rule; } });
var index_js_24 = require("./S6571/index.js"); // no-redundant-type-constituents
Object.defineProperty(exports, "S6571", { enumerable: true, get: function () { return index_js_24.rule; } });
var index_js_25 = require("./S6679/index.js"); // no-self-compare
Object.defineProperty(exports, "S6679", { enumerable: true, get: function () { return index_js_25.rule; } });
var index_js_26 = require("./S7060/index.js"); // no-self-import
Object.defineProperty(exports, "S7060", { enumerable: true, get: function () { return index_js_26.rule; } });
var index_js_27 = require("./S4327/index.js"); // no-this-alias
Object.defineProperty(exports, "S4327", { enumerable: true, get: function () { return index_js_27.rule; } });
var index_js_28 = require("./S3696/index.js"); // no-throw-literal
Object.defineProperty(exports, "S3696", { enumerable: true, get: function () { return index_js_28.rule; } });
var index_js_29 = require("./S6747/index.js"); // no-unknown-property
Object.defineProperty(exports, "S6747", { enumerable: true, get: function () { return index_js_29.rule; } });
var index_js_30 = require("./S1763/index.js"); // no-unreachable
Object.defineProperty(exports, "S1763", { enumerable: true, get: function () { return index_js_30.rule; } });
var index_js_31 = require("./S905/index.js"); // no-unused-expressions
Object.defineProperty(exports, "S905", { enumerable: true, get: function () { return index_js_31.rule; } });
var index_js_32 = require("./S1068/index.js"); // no-unused-private-class-members
Object.defineProperty(exports, "S1068", { enumerable: true, get: function () { return index_js_32.rule; } });
var index_js_33 = require("./S6676/index.js"); // no-useless-call
Object.defineProperty(exports, "S6676", { enumerable: true, get: function () { return index_js_33.rule; } });
var index_js_34 = require("./S6647/index.js"); // no-useless-constructor
Object.defineProperty(exports, "S6647", { enumerable: true, get: function () { return index_js_34.rule; } });
var index_js_35 = require("./S3504/index.js"); // no-var
Object.defineProperty(exports, "S3504", { enumerable: true, get: function () { return index_js_35.rule; } });
var index_js_36 = require("./S3498/index.js"); // object-shorthand
Object.defineProperty(exports, "S3498", { enumerable: true, get: function () { return index_js_36.rule; } });
var index_js_37 = require("./S6572/index.js"); // prefer-enum-initializers
Object.defineProperty(exports, "S6572", { enumerable: true, get: function () { return index_js_37.rule; } });
var index_js_38 = require("./S4138/index.js"); // prefer-for-of
Object.defineProperty(exports, "S4138", { enumerable: true, get: function () { return index_js_38.rule; } });
var index_js_39 = require("./S6598/index.js"); // prefer-function-type
Object.defineProperty(exports, "S6598", { enumerable: true, get: function () { return index_js_39.rule; } });
var index_js_40 = require("./S4156/index.js"); // prefer-namespace-keyword
Object.defineProperty(exports, "S4156", { enumerable: true, get: function () { return index_js_40.rule; } });
var index_js_41 = require("./S6606/index.js"); // prefer-nullish-coalescing
Object.defineProperty(exports, "S6606", { enumerable: true, get: function () { return index_js_41.rule; } });
var index_js_42 = require("./S6661/index.js"); // prefer-object-spread
Object.defineProperty(exports, "S6661", { enumerable: true, get: function () { return index_js_42.rule; } });
var index_js_43 = require("./S6666/index.js"); // prefer-spread
Object.defineProperty(exports, "S6666", { enumerable: true, get: function () { return index_js_43.rule; } });
var index_js_44 = require("./S6557/index.js"); // prefer-string-starts-ends-with
Object.defineProperty(exports, "S6557", { enumerable: true, get: function () { return index_js_44.rule; } });
var index_js_45 = require("./S3512/index.js"); // prefer-template
Object.defineProperty(exports, "S3512", { enumerable: true, get: function () { return index_js_45.rule; } });
var index_js_46 = require("./S1438/index.js"); // semi
Object.defineProperty(exports, "S1438", { enumerable: true, get: function () { return index_js_46.rule; } });
var index_js_47 = require("./S107/index.js"); // sonar-max-params
Object.defineProperty(exports, "S107", { enumerable: true, get: function () { return index_js_47.rule; } });
var index_js_48 = require("./S1534/index.js"); // sonar-no-dupe-keys
Object.defineProperty(exports, "S1534", { enumerable: true, get: function () { return index_js_48.rule; } });
var index_js_49 = require("./S131/index.js"); // switch-without-default
Object.defineProperty(exports, "S131", { enumerable: true, get: function () { return index_js_49.rule; } });
var index_js_50 = require("./S6535/index.js"); // unnecessary-character-escapes
Object.defineProperty(exports, "S6535", { enumerable: true, get: function () { return index_js_50.rule; } });
var index_js_51 = require("./S2688/index.js"); // use-isnan
Object.defineProperty(exports, "S2688", { enumerable: true, get: function () { return index_js_51.rule; } });
