// eslint-plugin-sonarjs@2.0.4/cjs/S6647/rule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.rule = void 0;
const index_js_1 = require("../core/index.js");
const decorator_js_1 = require("./decorator.js");
const index_js_2 = require("../helpers/index.js");
/**
 * Check if method with accessibility is not useless
 */
function checkAccessibility(node) {
    switch (node.accessibility) {
        case 'protected':
        case 'private':
            return false;
        case 'public':
            if (node.parent.type === 'ClassBody' &&
                'superClass' in node.parent.parent &&
                node.parent.parent.superClass) {
                return false;
            }
            break;
    }
    return true;
}
/**
 * Check if method is not useless due to typescript parameter properties and decorators
 */
function checkParams(node) {
    return !node.value.params.some(param => param.type === 'TSParameterProperty' || param.decorators?.length > 0);
}
/**
 * Check if the enclosing class is not decorated.
 */
function checkDecorator(node) {
    return !(node.parent.parent?.type === 'ClassDeclaration' && node.parent.parent.decorators?.length > 0);
}
/**
 * Check if the enclosing class does not inherit a protected constructor.
 */
function checkInheritance(node, context) {
    if (node.parent.type === 'ClassBody' &&
        'superClass' in node.parent.parent &&
        node.parent.parent.superClass) {
        const superClass = node.parent.parent.superClass;
        const variable = (0, index_js_2.getVariableFromName)(context, superClass.name, node);
        for (const def of variable?.defs ?? []) {
            if (def.type === 'ImportBinding') {
                return false;
            }
            if (def.node.type === 'ClassDeclaration') {
                const decl = def.node;
                if (decl.body.body.some(member => member.type === 'MethodDefinition' &&
                    member.kind === 'constructor' &&
                    member.accessibility === 'protected')) {
                    return false;
                }
            }
        }
    }
    return true;
}
const eslintNoUselessConstructor = index_js_1.eslintRules['no-useless-constructor'];
const originalRule = {
    meta: eslintNoUselessConstructor.meta,
    create(context) {
        const rules = eslintNoUselessConstructor.create(context);
        return {
            MethodDefinition(node) {
                if (node.value.type === 'FunctionExpression' &&
                    node.kind === 'constructor' &&
                    checkAccessibility(node) &&
                    checkParams(node) &&
                    checkDecorator(node) &&
                    checkInheritance(node, context)) {
                    rules.MethodDefinition(node);
                }
            },
        };
    },
};
exports.rule = (0, decorator_js_1.decorate)(originalRule);
