// eslint-plugin-sonarjs@2.0.4/cjs/S6544/rule.js
"use strict";
/*
 * SonarQube JavaScript Plugin
 * Copyright (C) 2011-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// https://sonarsource.github.io/rspec/#/rspec/S6544/javascript
Object.defineProperty(exports, "__esModule", { value: true });
exports.rule = void 0;
const index_js_1 = require("../typescript-eslint/index.js");
const index_js_2 = require("../core/index.js");
const index_js_3 = require("../helpers/index.js");
const meta_js_1 = require("./meta.js");
/**
 * We keep a single occurrence of issues raised by both rules, discarding the ones raised by 'no-async-promise-executor'
 * The current logic relies on the fact that the listener of 'no-misused-promises' runs first because
 * it is alphabetically "smaller", which is how we set them up in mergeRules.
 */
/**
 * start offsets of nodes that raised issues in typescript-eslint's no-misused-promises
 */
const flaggedNodeStarts = new Map();
const noMisusedPromisesRule = index_js_1.tsEslintRules['no-misused-promises'];
const decoratedNoMisusedPromisesRule = (0, index_js_3.interceptReport)(noMisusedPromisesRule, (context, descriptor) => {
    if ('node' in descriptor) {
        const node = descriptor.node;
        const start = node.range[0];
        if (!flaggedNodeStarts.get(start)) {
            flaggedNodeStarts.set(start, true);
            if (index_js_3.FUNCTION_NODES.includes(node.type)) {
                const loc = (0, index_js_3.getMainFunctionTokenLocation)(node, node.parent, context);
                context.report({ ...descriptor, loc });
            }
            else {
                context.report(descriptor);
            }
        }
    }
});
const noAsyncPromiseExecutorRule = index_js_2.eslintRules['no-async-promise-executor'];
const decoratedNoAsyncPromiseExecutorRule = (0, index_js_3.interceptReport)(noAsyncPromiseExecutorRule, (context, descriptor) => {
    if ('node' in descriptor) {
        const start = descriptor.node.range[0];
        if (!flaggedNodeStarts.get(start)) {
            context.report(descriptor);
        }
    }
});
exports.rule = {
    meta: (0, index_js_3.generateMeta)(meta_js_1.meta, {
        messages: {
            ...decoratedNoMisusedPromisesRule.meta.messages,
            ...decoratedNoAsyncPromiseExecutorRule.meta.messages,
        },
        hasSuggestions: true,
        schema: [
            {
                type: 'object',
                properties: {},
            },
        ],
    }),
    create(context) {
        return {
            'Program:exit': () => {
                flaggedNodeStarts.clear();
            },
            ...(0, index_js_3.mergeRules)(decoratedNoAsyncPromiseExecutorRule.create(context), decoratedNoMisusedPromisesRule.create(context)),
        };
    },
};
